pkg/timeline/service/appendMember.ts

```diff
@@ -0,0 +1,82 @@
+import { Ether, Result } from '@mikuroxina/mini-fn';
+import type { AccountID } from '../../accounts/model/account.js';
+import {
+  ListNotFoundError,
+  ListTooManyMembersError,
+  TimelineInsufficientPermissionError,
+} from '../model/errors.js';
+import type { List, ListID } from '../model/list.js';
+import { type ListRepository, listRepoSymbol } from '../model/repository.js';
+
+export class AppendListMemberService {
+  // ToDo: make this configurable
+  private LIST_MEMBER_LIMIT = 250;
+  constructor(private readonly listRepository: ListRepository) {}
+
+  /**
+   * @description Append member to list.
+   * NOTE: If account is already a member, returns error.
+   * NOTE: If list member count exceeds limit(250, static), returns error.
+   * @param listID
+   * @param accountID
+   * @param actorID
+   */
+  async handle(
+    listID: ListID,
+    accountID: AccountID,
+    actorID: AccountID,
+  ): Promise<Result.Result<Error, void>> {
+    const listRes = await this.listRepository.fetchList(listID);
+    if (Result.isErr(listRes)) {
+      return Result.err(
+        new ListNotFoundError('List not found', {
+          cause: Result.unwrapErr(listRes),
+        }),
+      );
+    }
+    const allowedRes = this.isAllowed(actorID, Result.unwrap(listRes));
+    if (Result.isErr(allowedRes)) {
+      return allowedRes;
+    }
+
+    const membersRes = await this.listRepository.fetchListMembers(listID);
+    if (Result.isErr(membersRes)) {
+      return Result.err(
+        new ListNotFoundError('List not found', {
+          cause: Result.unwrapErr(membersRes),
+        }),
+      );
+    }
+    const members = Result.unwrap(membersRes);
+
+    if (members.length >= this.LIST_MEMBER_LIMIT) {
+      return Result.err(
+        new ListTooManyMembersError('Too many members', {
+          cause: null,
+        }),
+      );
+    }
+
+    return await this.listRepository.appendListMember(listID, accountID);
+  }
+
+  private isAllowed(actor: AccountID, list: List): Result.Result<Error, void> {
```

Comment:
```suggestion
  private verifyActorPermission(actor: AccountID, list: List): Result.Result<Error, void> {
```
`isAllowed` だと `boolean` を返しそうなメソッド名に見えます. 検証を行ってその結果を `void` またはエラーで返すので, `verify` のような単語を使ってみてはどうでしょうか.